File: rust/P0001/sum-multiples/src/main.rs
use structopt::StructOpt;

mod solution;
use solution::sum_all_multiples_of_below;

/// Search for a pattern in a file and display the lines that contain it.
#[derive(StructOpt)]
struct Cli {
    a: u128,
    b: u128,
    below: u128
}

fn main() {
    let args = Cli::from_args();
    let ans = sum_all_multiples_of_below(args.a, args.b, args.below);
    println!("The sum of the multiples of {0} and {1} below {2} is: {3}", args.a, args.b, args.below, ans);
}

File: haskell/P0001/sum-multiples/app/Main.hs
module Main where

import Solution
import System.Environment (getArgs)

readMaybe :: Read a => String -> Maybe a
readMaybe s = case reads s of
                  [(val, "")] -> Just val
                  _           -> Nothing

printResult ::Int -> Int -> Int -> Int -> IO ()
printResult a b below ans = print ("The sum of the multiples of " ++ show a ++ " and " ++ show b ++ " below " ++ show below ++ " is: " ++ show ans)

printError :: IO ()
printError = putStrLn "error: exactly three positive integers needed"

main :: IO ()
main = do 
    args <- getArgs
    case args of
        [a, b, below] -> 
            case (ma, mb, mbelow) of
                (Just a, Just b, Just below)  -> printResult a b below (sumAllMultiplesOfBelow a b below)
                _                             -> printError
                where ma = readMaybe a :: Maybe Int
                      mb = readMaybe b :: Maybe Int
                      mbelow = readMaybe below :: Maybe Int
        _ -> printError